storage/replica.go

```diff
@@ -163,6 +163,7 @@ type Replica struct {
 	// proposeRaftCommandFn can be set to mock out the propose operation.
 	proposeRaftCommandFn func(cmdIDKey, roachpb.RaftCommand) <-chan error
 
+	readMu       sync.RWMutex    // Held during read-only commands
```

Comment:
`readMu` sounds confusingly similar to the `RLock` of the embedded `RWMutex`. Maybe call it `readOnlyCmdMu`?

I'd add more detail to the comment: "Held in read mode during read-only commands. Held in exclusive mode to prevent read-only commands from executing. Acquired before the embedded `RWMutex`"